src/OSPSuite.Core/Domain/Services/SimModelBatch.cs

```diff
@@ -13,66 +13,108 @@ public class SimModelBatch : SimModelManagerBase, ISimModelBatch
+         simModelExporter, simModelSimulationFactory)
       {
          _dataFactory = dataFactory;
-         _parameterValueCache = new Cache<string, double>();
       }
 
-      public void InitializeWith(IModelCoreSimulation modelCoreSimulation, IReadOnlyList<string> variableParameterPaths, bool calculateSensitivities = false, string simulationResultsName = null)
+      public IReadOnlyList<string> VariableParameterPaths { get; private set; }
+
+      public IReadOnlyList<string> VariableMoleculePaths { get; private set; }
+
+      public void InitializeWith(IModelCoreSimulation modelCoreSimulation, IReadOnlyList<string> variableParameterPaths,
+         IReadOnlyList<string> variableMoleculePaths, bool calculateSensitivities = false, string simulationResultsName = null)
       {
          _modelCoreSimulation = modelCoreSimulation;
-         _variableParameterPaths = variableParameterPaths;
          _simulationResultsName = simulationResultsName;
          _calculateSensitivities = calculateSensitivities;
-         _simModelSimulation = createAndFinalizeSimulation();
-         _allVariableParameters = _simModelSimulation.VariableParameters.ToList();
+         _simModelSimulation = createAndFinalizeSimulation(variableParameterPaths, variableMoleculePaths);
       }
 
       public void InitializeForSensitivity()
       {
-         InitializeWith(_modelCoreSimulation, _variableParameterPaths, calculateSensitivities: true, simulationResultsName: _simulationResultsName);
+         InitializeWith(_modelCoreSimulation, VariableParameterPaths, VariableMoleculePaths, calculateSensitivities: true,
+            simulationResultsName: _simulationResultsName);
       }
 
-      private Simulation createAndFinalizeSimulation()
+      private Simulation createAndFinalizeSimulation(IReadOnlyList<string> variableParameterPaths,
+         IReadOnlyList<string> variableMoleculePaths)
       {
          var simulationExport = CreateSimulationExport(_modelCoreSimulation, SimModelExportMode.Optimized);
          var simulation = CreateSimulation(simulationExport, x => x.CheckForNegativeValues = false);
-         setVariableParameters(simulation);
+         setVariableParameters(simulation, variableParameterPaths);
+         setVariableMolecules(simulation, variableMoleculePaths);
          FinalizeSimulation(simulation);
          return simulation;
       }
 
-      private void setVariableParameters(Simulation simulation) => SetVariableParameters(simulation, _variableParameterPaths, _calculateSensitivities);
+      private void setVariableParameters(Simulation simulation, IReadOnlyList<string> variableParameterPaths)
+      {
+         _allVariableParameters = SetVariableParameters(simulation, variableParameterPaths, _calculateSensitivities);
+         VariableParameterPaths = _allVariableParameters.Select(x => x.Path).ToList();
```

Comment:
I am settings them based on what simModel is actually returning because we are filtering non valid paths. I need make sure that I can check which parameters are `indeed` being varied by SimModel